### DISCUSSION TITLE
请对TWITTER_COOKIE的配置方式提供说明

### DISCUSSION THREAD

**Discussion by karegyudon** [DIYgod/RSSHub#14956] [Q&A] [answered]
### 这是一个什么样的功能？

https://docs.rsshub.app/routes/social-media#twitter 此处的文档提到：
Currently supports two authentication methods:

Using TWITTER_COOKIE (recommended): Configure the cookies of logged-in Twitter Web, at least including the fields auth_token and ct0. RSSHub will use this information to directly access Twitter's web API to obtain data.

但是找不到任何如何设置该变量的说明或者提示，能否给出个具体点的操作方式或者配置案例？

如使用username和password的方式，每隔几个小时就会收到twitter的提示邮件，有客户端进行了登陆，根据文档的提示，长期如此的情况下可能触发风控机制限制账户。

### 这个功能可以解决什么问题？

可以方便在自行部署上配置推荐的方式以访问twitter。

### 这不是重复的功能请求和 RSS 提案

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保这项功能尚未被请求及不是 [新的 RSS 提案](https://github.com/DIYgod/RSSHub/issues/new?assignees=&labels=RSS+proposal&template=rss_request_zh.yml)。

**Comment by assassinliujie** [comment, marked as answer]
在docker-compose.yml environment下面，TWITTER_COOKIE: 'auth_token=yourtoken; ct0=yourkey'

**Reply by FledgeXu** [reply]
这个是不是应该加到文档上……

**Reply by assassinliujie** [reply]
其实env中填cookie一般都是这么填吧

**Comment by aaro-n** [comment]
今天（3月29日）在RSSHub配置Twitter成功，记录下：
RSSHub运行平台及镜像：Fly.io ，diygod/rsshub:chromium-bundled
Fly.io平台分配的资源：2个CPU，512内存，shared-cpu-2x
Twitter环境变量：TWITTER_USERNAME，TWITTER_PASSWORD，TWITTER_AUTHENTICATION_SECRET，TWITTER_COOKIE
具体方法：
1.首先在fly.io运行rsshub，确保可以通过浏览器访问
2.将资源调整为2个cpu，1024内存，并使用`jlesage/firefox-esr:latest`登录Tweitter，这个主要作用是绕过人机识别，获取TWITTER_COOKIE变量所需要的内容。
3.根据RSSHub文档配置所需要的环境变量，并部署到fly.io 上，测试Twiitter，如果正常将资源调整为2个cpu，512mb内存。
参考：
[如何获取推特（twitter）账号的Token实现Token一键登录](https://goodcoin.medium.com/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E6%8E%A8%E7%89%B9-twitter-%E8%B4%A6%E5%8F%B7%E7%9A%84token%E5%AE%9E%E7%8E%B0token%E4%B8%80%E9%94%AE%E7%99%BB%E5%BD%95-d9f5b21f60da)

**Comment by aaro-n** [comment]
2024年7月17日，使用Twitter COOKIE环境变量变更如下：
# 环境变量变更如下：
`TWITTER_COOKIE=auth_token=1234567; ct0=aaaaaaaaaaaaaaaaaaaaaaa`
改为
`TWITTER_AUTH_TOKEN=1234567`
